SpatialGDK/Source/SpatialGDKTests/SpatialGDKEditor/SpatialGDKEditorSchemaGenerator/SpatialGDKEditorSchemaGeneratorTest.cpp

```diff
@@ -1,9 +1,836 @@
+	{
+		if (FString* ExpectedContentPtr = ExpectedContents.Find(CurrentClass->GetName()))
+		{
+			FString ExpectedContent = *ExpectedContentPtr;
+			ExpectedContent.ReplaceInline(TEXT("{{id}}"), *FString::FromInt(GetID()));
+			return (FileContent.Compare(ExpectedContent) == 0);
+		}
+		else
+		{
+			return false;
+		}
+	}
+
+private:
+	int GetID()
+	{
+		return freeId++;
+	}
+
+	int freeId = 10000;
+};
+
+void DeleteTestFolders()
+{
+	IPlatformFile& PlatformFile = FPlatformFileManager::Get().GetPlatformFile();
+	PlatformFile.DeleteDirectoryRecursively(*FPaths::Combine(FPaths::ProjectContentDir(), TEXT("Spatial/Tests/")));
+	PlatformFile.DeleteDirectoryRecursively(*gSchemaOutputFolder);
+}
+
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_spatial_type_class_WHEN_checked_if_supported_THEN_is_supported)
+{
+	// GIVEN
+	const UClass* SupportedClass = USpatialTypeObjectStub::StaticClass();
+
+	// WHEN
+	bool bIsSupported = SpatialGDKEditor::Schema::IsSupportedClass(SupportedClass);
+
+	// THEN
+	TestTrue("Spatial type class is supported", bIsSupported);
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_class_derived_from_spatial_type_class_WHEN_checked_if_supported_THEN_is_supported)
+{
+	// GIVEN
+	const UClass* SupportedClass = UChildOfSpatialTypeObjectStub::StaticClass();
+
+	// WHEN
+	bool bIsSupported = SpatialGDKEditor::Schema::IsSupportedClass(SupportedClass);
+
+	// THEN
+	TestTrue("Child of a Spatial type class is supported", bIsSupported);
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_null_pointer_WHEN_checked_if_supported_THEN_is_not_supported)
+{
+	// GIVEN
+	const UClass* SupportedClass = nullptr;
+
+	// WHEN
+	bool bIsSupported = SpatialGDKEditor::Schema::IsSupportedClass(SupportedClass);
+
+	// THEN
+	TestFalse("Null pointer is not a valid argument", bIsSupported);
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_non_spatial_type_class_WHEN_checked_if_supported_THEN_is_not_supported)
+{
+	// GIVEN
+	const UClass* SupportedClass = UNotSpatialTypeObjectStub::StaticClass();
+
+	// WHEN
+	bool bIsSupported = SpatialGDKEditor::Schema::IsSupportedClass(SupportedClass);
+
+	// THEN
+	TestFalse("Non spatial type class is not supported", bIsSupported);
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_class_derived_from_non_spatial_type_class_WHEN_checked_if_supported_THEN_is_not_supported)
+{
+	// GIVEN
+	const UClass* SupportedClass = UChildOfNotSpatialTypeObjectStub::StaticClass();
+
+	// WHEN
+	bool bIsSupported = SpatialGDKEditor::Schema::IsSupportedClass(SupportedClass);
+
+	// THEN
+	TestFalse("Child of Non-Spatial type class is not supported", bIsSupported);
+	return true;
+}
+
+SCHEMA_GENERATOR_TEST(GIVEN_a_class_with_not_spatial_tag_WHEN_checked_if_supported_THEN_is_not_supported)
+{
+	// GIVEN
+	const UClass* SupportedClass = UNotSpatialTypeObjectStub::StaticClass();
+
+	// WHEN
+	bool bIsSupported = SpatialGDKEditor::Schema::IsSupportedClass(SupportedClass);
+
+	// THEN
+	TestFalse("Class with Not Spatial flag is not supported", bIsSupported);
```

Comment:
```suggestion
	TestFalse("Class with Not Spatial Type flag is not supported", bIsSupported);
```

Reply:
"Class with Not Spatial flag is not supported"

Excuse my ignorance here, how are these strings used, are they output as the test result?

Reply:
Yes, in log you would get an output of kind `"Expected 'Class with Not Spatial Type flag is not supported' to be false."`

Reply:
In case there is an error.